### DISCUSSION TITLE
command line inexact?

### DISCUSSION THREAD

**Discussion by anthonyfca** [minio/minio#13273] [Q&A] [answered]
this is progress after 7 hours of trying, basically random changes as the docs don't explain (after many many hours of reading, I suspect they only make sense if you know the answers, so hopefully this will be useful feedback - just before I lose the will to live):

None of the ip addresses work and all get the 

minitv:/ tv$ minio server ~/NODE./Volumes/DISK{1...4}/data
Formatting 1st pool, 1 set(s), 4 drives per set.
WARNING: Host local has more than 2 drives of set. A host failure will result in data becoming unavailable.
Automatically configured API requests per node based on available memory on the system: 78
Status:         4 Online, 0 Offline. 
API: http://10.10.0.116:9000  http://10.10.0.63:9000  http://127.0.0.1:9000                   
RootUser: minio 
RootPass: miniostorage 

Console: http://10.10.0.116:54281 http://10.10.0.63:54281 http://127.0.0.1:54281          
RootUser: minio 
RootPass: miniostorage 

Command-line: https://docs.min.io/docs/minio-client-quickstart-guide
   $ mc alias set myminio http://10.10.0.116:9000 minio miniostorage

Documentation: https://docs.min.io

WARNING: Console endpoint is listening on a dynamic port (54281), please use --console-address ":PORT" to choose a static port.

I am trying to get a "distributed" system working on 4 local disks so i can use the worm tech.
Any help will save me from all kinds of unpleasantness
Not getting any work done today!
Anthony

**Comment by krisis** [comment, marked as answer]
To deploy MinIO in distributed mode you need to run the following command on all the 4 servers,

`minio server http://server1/NODE./Volumes/DISK{1...4}/data http://server2/NODE./Volumes/DISK{1...4}/data http://server3/NODE./Volumes/DISK{1...4}/data http://server4/NODE./Volumes/DISK{1...4}/data`

This assumes that your servers are named `server1`, `server2`, `server3` and `server4`. Please change the command line to reflect your server names (or IP addresses).

**Comment by anthonyfca** [comment]
My “servers” are four local hard disks 

How do I name them?

Indeed given they are DAS odds how do I give them an address?

Thanks for your reply - Anthony

**Comment by krisis** [comment]
@anthonyfca ok. Then you wish to deploy MinIO in a single node erasure coded setup. In that case, your command line seems fine as it were. 
Did this work for you? `mc alias set myminio http://10.10.0.116:9000 minio miniostorage`. What was the error you saw, if this didn't work?
Could you paste the error message here?

**Comment by anthonyfca** [comment]
It seemed to work and an S3 backup worked but then I found I could simply delete files when on that machine ; I thought w o r m meant that’s not possible

**Comment by anthonyfca** [comment]
I was trying for distributed mode to get immutability, hence ; disks

**Comment by krisis** [comment]
How did you set up worm on MinIO? How did you try to delete the files?

**Comment by ravindk89** [comment]
@anthonyfca you can use local hard drives, you need to mount them using sequential drive names. So if you have:

```
/dev/sdb
/dev/sdc
/dev/sdd
/dev/sde
```

You would mount that as something like

```
/mnt/disk1
/mnt/disk2
/mnt/disk3
/mnt/disk4
```

At which point you can do

```
minio server http://<hostname>/mnt/disk{1...4}
```

Assuming DAS = direct attached storage, all that is required is mounting those drives. You can do this via `mount` or using fstab. If you are using fstab, we'd recommend using disk UUIDs to associate a disk to a given mount point.

The general syntax is documented [here](https://docs.min.io/minio/baremetal/installation/deploy-minio-distributed.html), along with more guidance on setting drives and such. 

Once we validate that you have a distributed MinIO setup, we can focus on WORM locking - we have documentation [here](https://docs.min.io/minio/baremetal/object-retention/minio-object-locking.html) that includes guidance on configuring WORM. Because of how versioning works, you might be seeing deletes "succeed" - but what is happening is you are creating a delete marker for an object that is WORM locked (as is S3 behavior). Attempting a versioned delete on any WORM locked object *should* fail.

**Comment by anthonyfca** [comment]
Hi, yes (good morning) I have mounted my disks as 

/VOLUMEDISK1
/VOLUMEDISK2
/VOLUMEDISK3
/VOLUMEDISK4

Yes understood has to be “distributed” - whether I can do that with just 4 disks is not clear to me, though I seem to have managed it in that I get an Arq response saying files cannot be deleted when it tries to update an index file. 

(macOS hence /volume)

DAS is directly attached storage (as opposed to NAS)

I use ~/node/volumes (no dot in my final) /DISK[1…4] 

I don’t know what node is supposed to mean.

(I find the docs never give me the complete story to understand - been three weeks now.)
Two days ago I finally got the gui up using that command with node in it.

When it says ‘run command on each server”, given this is on ONE Mac mini I am not at all clear how I have set up a server at all, let alone 4. 

It looks like it is very simple if only one understand the commands - as illustrated by me not being clear what I have managed to set up, I THINK it is distributed, across 4  five TB disks. 

Thanks for the docs link I’ll read those assuming it’s not docs I have seen before and spent hours reading - asking for help is a last resort, but clearly I have no idea what I am doing.

Thanks both it’s lovely to have some help - Anthony
